djhtml/__main__.py

```diff
@@ -1,210 +1,152 @@
-
-        $ djhtml -i file1.html file2.html
+    changed_files = 0
+    unchanged_files = 0
+    problematic_files = 0
 
-    """
-    target_extension = ".html"
+    suffixes = [".html"]
     Mode = modes.DjHTML
     if sys.argv[0].endswith("djtxt"):
         Mode = modes.DjTXT
-        target_extension = ".txt"
+        suffixes = [".txt"]
     if sys.argv[0].endswith("djcss"):
         Mode = modes.DjCSS
-        target_extension = ".css"
+        suffixes = [".css", ".scss"]
     if sys.argv[0].endswith("djjs"):
         Mode = modes.DjJS
-        target_extension = ".js"
+        suffixes = [".js"]
 
-    changed_files = 0
-    unchanged_files = 0
-    problematic_files = 0
+    if len(options.input_filenames) > 1 and "-" in options.input_filenames:
+        sys.exit("I’m sorry Dave, I’m afraid I can’t do that.")
 
-    parser = argparse.ArgumentParser(
-        description=(
-            "DjHTML is a fully automatic template indenter that works with mixed"
-            " HTML/CSS/Javascript templates that contain Django or Jinja template"
-            " tags. It works similar to other code-formatting tools such as Black and"
-            " interoperates nicely with pre-commit. Full documentation can be found at"
-            " https://github.com/rtts/djhtml"
-        ),
-    )
-    parser.add_argument(
-        "-i", "--in-place", action="store_true", help="modify files in-place"
-    )
-    parser.add_argument("-c", "--check", action="store_true", help="don't modify files")
-    parser.add_argument("-q", "--quiet", action="store_true", help="be quiet")
-    parser.add_argument(
-        "-t",
-        "--tabwidth",
-        metavar="N",
-        type=int,
-        default=4,
-        help="tabwidth (default is 4)",
-    )
-    parser.add_argument(
-        "-o",
-        "--output-file",
-        metavar="filename",
-        default="-",
-        help="output filename",
-    )
-    parser.add_argument(
-        "input_filenames",
-        metavar="filenames",
-        nargs="*",
-        default=["-"],
-        help="input filenames (either paths or directories)",
-    )
-    parser.add_argument("-d", "--debug", action="store_true", help=argparse.SUPPRESS)
-    args = parser.parse_args()
-
-    if args.in_place and "-" in args.input_filenames:
-        sys.exit("I’m sorry Dave, I’m afraid I can’t do that")
-
-    if len(args.input_filenames) > 1 and not args.in_place and not args.check:
-        sys.exit("Will not modify files in-place without -i option")
-
-    for input_filename in _generate_files(args.input_filenames, target_extension):
+    for filename in _generate_filenames(options.input_filenames, suffixes):
         # Read input file
         try:
-            input_file = (
-                sys.stdin if input_filename == "-" else open(input_filename, "r")
-            )
+            input_file = sys.stdin if filename == "-" else open(filename, "r")
             source = input_file.read()
+            input_file.close()
```

Comment:
Here, problematic files will no longer be closed. I suspect that's not intentional?

Reply:
Indeed. I went to great length to avoid a `with open()` statement as to preserve the fancy ternary operator. I'll give up and just do it properly instead :pray: